src/contracts/core/AllocationManager.sol

```diff
@@ -430,25 +350,43 @@ contract AllocationManager is
         IStrategy strategy,
         uint16 numToComplete,
         uint192 nextIndex
-    ) internal view returns (uint64 freeMagnitudeToAdd, uint192 completed) {
-        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+    ) internal returns (uint64 freeMagnitudeToAdd, uint192 completed) {
+        uint256 numDeallocations = _pendingDeallocationOperatorSets[operator][strategy].length;
         freeMagnitudeToAdd = 0;
-        while (nextIndex < pendingFreeMagnitudeLength && completed < numToComplete) {
-            PendingFreeMagnitude memory pendingFreeMagnitude = _pendingFreeMagnitude[operator][strategy][nextIndex];
-            // pendingFreeMagnitude is ordered by completableTimestamp. If we reach one that is not completable yet, then break
+        while (nextIndex < numDeallocations && completed < numToComplete) {
+            bytes32 opsetKey = _pendingDeallocationOperatorSets[operator][strategy][nextIndex];
+            MagnitudeInfo memory opsetMagnitudeInfo = _operatorMagnitudeInfo[operator][strategy][opsetKey];
+            // _pendingDeallocationOperatorSets is ordered by `effectTimestamp`. If we reach one that is not completable yet, then break
             // loop until completableTimestamp is < block.timestamp
-            if (block.timestamp < pendingFreeMagnitude.completableTimestamp) {
+            if (block.timestamp < opsetMagnitudeInfo.effectTimestamp) {
                 break;
             }
 
             // pending free magnitude can be added to freeMagnitude
-            freeMagnitudeToAdd += pendingFreeMagnitude.magnitudeDiff;
+            freeMagnitudeToAdd += uint64(-opsetMagnitudeInfo.pendingMagnitudeDiff);
+            _operatorMagnitudeInfo[operator][strategy][opsetKey] = MagnitudeInfo({
+                currentMagnitude: opsetMagnitudeInfo.currentMagnitude - uint64(-opsetMagnitudeInfo.pendingMagnitudeDiff),
+                pendingMagnitudeDiff: 0,
+                effectTimestamp: 0
+            });
+
             ++nextIndex;
             ++completed;
         }
         return (freeMagnitudeToAdd, completed);
     }
 
+    function _calculateNewCurrentMagnitude(MagnitudeInfo memory magnitudeInfo) internal returns (int64) {
+        int64 newCurrentMagnitude;
+        if (magnitudeInfo.pendingMagnitudeDiff >= 0) {
+            magnitudeInfo.currentMagnitude = magnitudeInfo.currentMagnitude + uint64(magnitudeInfo.pendingMagnitudeDiff);
+        } else {
+            magnitudeInfo.currentMagnitude = magnitudeInfo.currentMagnitude - uint64(-magnitudeInfo.pendingMagnitudeDiff);
+        }
```

Comment:
This can't be right :grin: 

We're always returning zero.

To be clear - obviously forgot to actually set `newCurrentMagnitude` here, but do we actually want to update `magnitudeInfo.currentMagnitude`? If so, let's make it clear that this will modify the struct by changing the method name somehow

Reply:
Looks like you pushed a fix here, much clearer. Resolving